### ISSUE TITLE
API ignores redirection errors when user is not authenticated

### ISSUE THREAD

**Issue by GeoffWilliams** [puppetlabs/puppet-classify#5]
## Overview

If you connect to the node classifier API using unauthorised certificates you will receive a ruby error from puppet-classify.
## Expected result

When connecting with unauthorised certificates, puppet-classify should alert the user to this by displaying an error message and exiting
## Actual result

When requests are made with an unauthorized certificate, the node classifier will respond with a JSON redirect message similar to:

``` ruby
{"redirect"=>"/classifier-api/v1/groups", "msg"=>"Route requires authentication", "kind"=>"puppetlabs.rbac/user-unauthenticated"}
```

The redirect instruction is not processed and puppet-classify continues processing which results in a ruby error:

```
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `[]': no implicit conversion of String into Integer (TypeError)
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `block in get_group_id'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `each'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `find'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `get_group_id'
    from classify_pe_master_group.rb:86:in `<main>'
```

**Comment by briancain** [comment]
@GeoffWilliams do you have an example that will result in that stack trace? I attempted to use certs in PE that are not authenticated, and I just received the redirect error hash but no stack trace.

```
[root@athtkfcdu5l0obi ~]# cat wrongcerts.rb
require 'puppetclassify'
# URL of classifier as well as certificates and private key for auth
auth_info = {
  "ca_certificate_path" => "/etc/puppetlabs/puppet/ssl/certs/ca.pem",
  #"certificate_path"    => "/etc/puppetlabs/puppet/ssl/certs/athtkfcdu5l0obi.delivery.puppetlabs.net.pem",
  "certificate_path"    => "/etc/puppetlabs/puppet/ssl/certs/pe-internal-classifier.pem",
  #"private_key_path"    => "/etc/puppetlabs/puppet/ssl/private_keys/athtkfcdu5l0obi.delivery.puppetlabs.net.pem"
  "private_key_path"    => "/etc/puppetlabs/puppet/ssl/private_keys/pe-internal-classifier.pem"
}

classifier_url = 'https://athtkfcdu5l0obi.delivery.puppetlabs.net:4433/classifier-api'
puppetclassify = PuppetClassify.new(classifier_url, auth_info)
# Get all the groups
puts puppetclassify.groups.get_groups
[root@athtkfcdu5l0obi ~]# /opt/puppetlabs/puppet/bin/ruby wrongcerts.rb
{"redirect"=>"/classifier-api/v1/groups", "msg"=>"Route requires authentication", "kind"=>"puppetlabs.rbac/user-unauthenticated"}
```

**Comment by briancain** [comment]
@GeoffWilliams Ah, from your stack trace, it looks like you're attempting to use the function `get_group_id`. In that case, I can reproduce the error:

```
[root@athtkfcdu5l0obi ~]# /opt/puppetlabs/puppet/bin/ruby wrongcerts.rb
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `[]': no implicit conversion of String into Integer (TypeError)
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `block in get_group_id'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `each'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `find'
    from /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/puppetclassify-0.1.0/lib/puppetclassify/groups.rb:26:in `get_group_id'
    from wrongcerts.rb:14:in `<main>'
```

Yep, totally a bug. I'll try to get a fix out as soon as I can. Thanks for reporting it!

**Comment by briancain** [comment]
@GeoffWilliams Addressed in https://github.com/puppetlabs/puppet-classify/pull/6. Will merge soon and cut a gem release which should be 0.1.1.

**Comment by briancain** [comment]
Fixed in 315d7d15e3e8537baa1b510bf454c8046f7684d5
